src/thor/bidirectional_astar.cc

```diff
@@ -524,10 +532,31 @@ BidirectionalAStar::GetBestPath(valhalla::Location& origin,
 
   // we use a non varying time for all time dependent routes until we can figure out how to vary the
   // time during the path computation in the bidirectional algorithm
-  bool invariant = options.date_time_type() != Options::no_time;
+  bool invariant = options.date_time_type() == Options::invariant;
+  bool arrive_by = options.date_time_type() == Options::arrive_by;
+
   // Get time information for forward and backward searches
-  auto forward_time_info = TimeInfo::make(origin, graphreader, &tz_cache_);
-  auto reverse_time_info = TimeInfo::make(destination, graphreader, &tz_cache_);
+  // and determine which strategy to use on the far end
+  TimeInfo forward_time_info;
+  TimeInfo reverse_time_info;
+
+  if (arrive_by) {
+    reverse_time_info = TimeInfo::make(destination, graphreader, &tz_cache_);
+    forward_time_info =
+        reverse_time_tracking == Options_ReverseTimeTracking_heuristic
+            ? EstimateReverseStartTime(graphreader, origin, destination, kReverseTTHeuristicFactor,
+                                       reverse_time_info, costing_, arrive_by)
+            : TimeInfo::invalid();
+  } else {
+    forward_time_info = TimeInfo::make(origin, graphreader, &tz_cache_);
+    reverse_time_info = invariant
+                            ? TimeInfo::make(destination, graphreader, &tz_cache_)
+                            : (reverse_time_tracking == Options_ReverseTimeTracking_heuristic
+                                   ? EstimateReverseStartTime(graphreader, origin, destination,
+                                                              kReverseTTHeuristicFactor,
+                                                              forward_time_info, costing_, arrive_by)
+                                   : TimeInfo::invalid());
+  }
```

Comment:
on the "forward" expansion simply make a regular timeinfo object, and on the other one either
  - make a regular one if invariant
  - make in invalid one 
  - or make one based on a heuristic